src/helm/clients/vertexai_client.py

```diff
@@ -346,23 +345,36 @@ def complete_for_valid_error(error_message: str) -> RequestResult:
             try:
 
                 def do_it() -> Dict[str, Any]:
-                    raw_response = model.generate_content(
+                    response = model.generate_content(
                         contents, generation_config=parameters, safety_settings=self.safety_settings
                     )
-                    if raw_response._raw_response.prompt_feedback.block_reason != 0:
-                        hlog(f"Content blocked for prompt: {request.multimodal_prompt}")
-                        return {"error": self.CONTENT_POLICY_VIOLATED_FINISH_REASON}
+                    candidates: List[Candidate] = response.candidates
 
-                    return {"predictions": [{"text": raw_response.candidates[0].text}]}
+                    if not candidates:
+                        raise VertexAIContentBlockedError("No candidates in response due to content blocking")
+
+                    # We should only have one candidate
+                    candidate: Candidate = candidates[0]
+                    if (
+                        candidate.finish_reason in VertexAIChatClient.CONTENT_BLOCKED_FINISH_REASONS
+                        or not candidate.content.parts
+                    ):
+                        # The prediction was either blocked due to safety settings or the model stopped and returned
+                        # nothing (which also happens when the model is blocked).
+                        # For now, we don't cache blocked requests, because we are trying to get the
+                        # content blocking removed.
+                        raise VertexAIContentBlockedError("Content has no parts due to content blocking")
+
+                    return {"predictions": [{"text": candidate.text}]}
 
                 raw_cache_key = {"model_name": model_name, "prompt": prompt_key, **parameters}
                 if completion_index > 0:
                     raw_cache_key["completion_index"] = completion_index
 
                 cache_key = CachingClient.make_cache_key(raw_cache_key, request)
                 response, cached = self.cache.get(cache_key, wrap_request_time(do_it))
-            except (requests.exceptions.RequestException, ValueError) as e:
-                if str(e) == self.CONTENT_HAS_NO_PARTS_ERROR:
+            except (requests.exceptions.RequestException, VertexAIContentBlockedError) as e:
+                if "Content has no parts" in str(e):
```

Comment:
Don't do this, it is brittle and will break if the Vertex AI API changes their error string. Is it sufficient to just do `except VertexAIContentBlockedError:`?